### ISSUE TITLE
Support unloaded enums and classes in Intellisense

### ISSUE THREAD

**Issue by cdelcambre** [PowerShell/vscode-powershell#2312]
VSCode Version: 1.40.1
OS Version: Windows 10, 1903
Powershell Extension: 2019.11.0

Steps to Reproduce:

Create an Enum
Create a function which references that Enum as a parameter.
Call that function, there will be no autosense when populating the parameters and values for said function.

Simple Example
```powershell
Enum AzureSqlEdition
{
Standard
Premium
Basic
}

function functAutoSenseBroke
{
param
(
[Parameter(Mandatory=$True)][AzureSqlEdition]$DbType,
[Parameter(Mandatory=$True)][string]$OtherParameter
)
Write-Host $DbType
Write-Host $OtherParameter
}

function functAutoSenseWorks
{
param ( [Parameter(Mandatory=$True)][string]$DbType )
Write-Host $DbType
}
#works but no autosense
functAutoSenseBroke -DbType ([AzureSqlEdition]::Standard) -OtherParameter "other"

#Simple Function, no Enum, autosense works, for comparison purposes
functAutoSenseWorks -DbType "test"
```

**Comment by SydneyhSmith** [comment]
@cdelcambre thanks for opening this issue, and providing repro steps! I want to ensure that I understand what intellisense you are looking for/expecting, because I am having some trouble reproducing the issue
![image](https://user-images.githubusercontent.com/43417619/69102615-cabb0080-0a17-11ea-822b-0dafa1575889.png)
![image](https://user-images.githubusercontent.com/43417619/69102713-1a013100-0a18-11ea-8e4d-8b178695ae57.png)
![image](https://user-images.githubusercontent.com/43417619/69102667-f1793700-0a17-11ea-9dd5-b462054bda9e.png)

It would be great if you could attach a gif/screenshot or a description of what intellisense you are looking for--thanks!

**Comment by cdelcambre** [comment]
Thank you for looking into this @SydneyhSmith!  Your screen shots are exactly what I am hoping for but I don't get what you get on the function functAutoSenseBroke.  For both I get a long list of nonsense parameter name suggestions I could definitely live without.   Typing the text of the parameter I am looking for narrows down the noise.  Perhaps that's a symptom of my problem because I don't see that long list of noise suggestions in your screen shots.  Maybe I have a config issue?  This is a clean install done today of both VS code and PowerShell extension.  All default settings.

![image](https://user-images.githubusercontent.com/28201436/69103839-56825c00-0a1b-11ea-9ee9-2d6595d3320b.png)

![image](https://user-images.githubusercontent.com/28201436/69104008-d14b7700-0a1b-11ea-96af-5c7c107b3e5e.png)

Noise!
![image](https://user-images.githubusercontent.com/28201436/69104634-d1e50d00-0a1d-11ea-8f55-bb430782807e.png)

Extension Details
![image](https://user-images.githubusercontent.com/28201436/69104683-05279c00-0a1e-11ea-9998-79e07df43e85.png)

VS Code Details:
Version: 1.40.1 (user setup)
Commit: 8795a9889db74563ddd43eb0a897a2384129a619
Date: 2019-11-13T16:49:35.976Z
Electron: 6.1.2
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Windows_NT x64 10.0.18362

**Comment by DarkLite1** [comment]
I reported a similar issue in #2117 . For arguments sake I tried your example and have the same issue, there simply is no argument or parameter completion on the function `Get-Stuff`.

```powershell
Enum AzureSqlEdition {
    Standard
    Premium
    Basic
}

Function Get-Stuff {
    [CmdLetBinding()]
    Param (
        [Parameter(Mandatory)]
        [AzureSqlEdition]$DbType,
        [Parameter(Mandatory)]
        [string]$OtherParameter
    )

    $PSBoundParameters
}
```
The completion is simply not done when typing partially `-db` to get the full parameter name `-DbType`. Even thought the suggestion on top says it is known:

![image](https://user-images.githubusercontent.com/10828570/69154696-c4448b80-0ae0-11ea-9786-3e28e2dcce80.png)

**Comment by SeeminglyScience** [comment]
Yeah completion acts differently when the source is also in the text it's trying to complete for.  If you:

1. Copy the whole script
1. Paste it into a PSReadLine prompt without hitting enter or running any of the lines
1. Put your cursor on the same spot and press tab

Then it'll probably act same.  A very similar issue is PowerShell/PowerShell#10567.  This is something that needs to be corrected in PowerShell itself (the engine provides the API that does all of vscode-powershell's completions)

**Comment by SydneyhSmith** [comment]
Closing as an external fix in PowerShell  PowerShell/PowerShell#10567
